examples/awp_aws_onboarding/main.tf

```
@@ -0,0 +1,714 @@
+					"kms:PutKeyPolicy",
+					"kms:ScheduleKeyDeletion",
+					"kms:CancelKeyDeletion",
+				]
+				Resource = "*"
+			},
+			{
+				Sid       = "Allow attachment of persistent resources"
+				Effect    = "Allow"
+				Principal = {
+					AWS = "arn:${data.aws_partition.current.partition}:iam::${local.cloud_guard_backend_account_id}:root"
+				}
+				Action = [
+					"kms:CreateGrant",
+					"kms:ListGrants",
+					"kms:RevokeGrant",
+				]
+				Resource = "*"
+				Condition = {
+					Bool = {
+						"kms:GrantIsForAWSResource" = true
+					}
+				}
+			},
+		]
+	})
+}
+#END AWP MR key for snapshot re-encryption
+
+# The CloudGuardAWPKeyAlias resource defines a KMS key alias that is used to reference the KMS key in SaaS mode.
+resource "aws_kms_alias" "CloudGuardAWPKeyAlias" {
+	count      = local.scan_mode == "saas" ? 1 : 0
+	name       = "alias/CloudGuardAWPKey"
+	target_key_id = aws_kms_key.CloudGuardAWPKey[count.index].arn
+	depends_on = [
+		aws_kms_key.CloudGuardAWPKey
+	]
+}
+
+# The dome9_awp_aws_onboarding resource defines a Dome9 AWP AWS Onboarding.
+# The Dome9 AWP AWS Onboarding resource allows you to onboard an AWS account to Dome9 AWP.
+# The cloudguard_account_id attribute is used to specify the CloudGuard account id of the AWS account.
+# The cross_account_role_name attribute is used to specify the name of the cross account role that is used to allow AWP to access the AWS account.
+# The cross_account_role_external_id attribute is used to specify the external id of the cross account role that is used to allow AWP to access the AWS account.
+# The scan_mode attribute is used to specify the scan mode of the Dome9 AWP AWS Onboarding. The valid values are "inAccount" and "saas".
+# The agentless_account_settings attribute is used to specify the agentless account settings of the Dome9 AWP AWS Onboarding.
+# The disabled_regions attribute is used to specify the disabled regions of the agentless account settings of the Dome9 AWP AWS Onboarding.
+# The scan_machine_interval_in_hours attribute is used to specify the scan machine interval in hours of the agentless account settings of the Dome9 AWP AWS Onboarding.
+# The max_concurrence_scans_per_region attribute is used to specify the max concurrence scans per region of the agentless account settings of the Dome9 AWP AWS Onboarding.
+# The skip_function_apps_scan attribute is used to specify whether to skip the function apps scan of the agentless account settings of the Dome9 AWP AWS Onboarding.
+# The custom_tags attribute is used to specify the custom tags of the agentless account settings of the Dome9 AWP AWS Onboarding.
+resource "dome9_awp_aws_onboarding" "awp_aws_onboarding_test" {
+	cloudguard_account_id = "CLOUDGUARD_ACCOUNT_ID or EXTERNAL_AWS_ACCOUNT_NUMBER"
+	cross_account_role_name = aws_iam_role.CloudGuardAWPCrossAccountRole.name
+	cross_account_role_external_id = local.cross_account_role_external_id
+	scan_mode = local.scan_mode
+	agentless_account_settings {
```

the setting are optional, change that to this example:
```suggestion
	  # Optional Settings
  # e.g:
	#   agentless_account_settings { 					
	#     disabled_regions                 = []   // e.g: ["us-west-1", "us-west-2"]
	#     scan_machine_interval_in_hours   = 24
	#     max_concurrence_scans_per_region = 20
	#     skip_function_apps_scan          = false 
	#     custom_tags = {
	#       # example of custom tags
	#       tag1 = "value1"
	#       tag2 = "value2"
	#       tag3 = "value3"
	#     }
	#   }
```

---

done